x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java

```diff
@@ -212,6 +209,56 @@ private void collectStatsForTransformsWithoutTasks(Request request,
             }
         );
 
-        dataFrameTransformsConfigManager.getTransformStats(transformsWithoutTasks, searchStatsListener);
+        dataFrameTransformsConfigManager.getTransformStoredDoc(transformsWithoutTasks, searchStatsListener);
+    }
+
+    private void populateSingleStoppedTransformStat(DataFrameTransformStoredDoc transform,
+                                                    ActionListener<DataFrameTransformCheckpointingInfo> listener) {
+        transformsCheckpointService.getCheckpointStats(transform.getId(), transform.getTransformState().getCheckpoint(),
+            transform.getTransformState().getCheckpoint() + 1, transform.getTransformState().getIndexerState(),
+            transform.getTransformState().getPosition(), transform.getTransformState().getProgress(),
+            ActionListener.wrap(
+                listener::onResponse,
+                e -> {
+                    logger.warn("Failed to retrieve checkpointing info for transform [" + transform.getId() + "]", e);
+                    listener.onResponse(DataFrameTransformCheckpointingInfo.EMPTY);
+                }
+            ));
+    }
+
+    private void addCheckpointingInfoForTransformsWithoutTasks(List<DataFrameTransformStats> allStateAndStats,
+                                                               List<DataFrameTransformStoredDoc> statsForTransformsWithoutTasks,
+                                                               ActionListener<Void> listener) {
+
+        if (statsForTransformsWithoutTasks.isEmpty()) {
+            // No work to do, but we must respond to the listener
+            listener.onResponse(null);
```

Comment:
missing `return` after the response. Meaning this method continues to execute even after the listener is notified.